src/Factories/PDOFactory.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+
+namespace App\Factories;
+
+
+use PDO;
+use Psr\Container\ContainerInterface;
+
+class PDOFactory
+{
+    public function __invoke(ContainerInterface $container): PDO
+    {
+        $settings = $container->get('settings')['db'];
+        $db = new PDO($settings['host'] . $settings['name'], $settings['user'], $settings['password']);
+        $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+        if ($settings['debug']) {
+            $db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+        }
+        return $db;
+    }
+}
```

Comment:
i think it's good having this factory as a class - maybe we can make it so that all factories in the skeleton are classes? that way it would match the portal code (for the most part) and also make it so that we only have to explain it one way rather than both ways?

Reply:
I like this idea, could make factory classes for the renderer and the logger as well